### PR TITLE
Fixed stacking cables of different types

### PR DESCRIPTION
**Pull request by JohnGinnane**
After a little testing it seems neither humans nor cyborgs could stack a
"random" cable (/obj/item/stack/cable_coil/random/) onto a normal type
(/obj/item/stack/cable_coil) or cyborg table
(/obj/item/stack/cable_coil/cyborg). I've made it so it checks if both
the target and the weapon are the same "type" in another method.

I've also added a little message to let the non-robot know that the
stack is already at maximum capacity when they try to add something to
the stack. (e.g. 30 units for cable coils)

### CODE REVIEW

**Comment by JohnGinnane** [comment]
This is a fix for https://github.com/HippieStationCode/HippieStation13/issues/1879

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -236,8 +236,13 @@
 	return
 
 /obj/item/stack/attackby(obj/item/W, mob/user, params)
-	if(istype(W, src.type))
+	if(istype(W, src.type) || (istype(W, /obj/item/stack/cable_coil) && istype(src, /obj/item/stack/cable_coil)))
```
Comment (JohnGinnane) [inline code/game/objects/items/stacks/stack.dm]: It seems as though if W type was `/obj/item/stack/cable_coil` or `/obj/item/stack/cable_coil/cyborg` and src type was `/obj/item/stack/cable_coil/random` it wouldn't match

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -236,8 +236,13 @@
...
 		var/obj/item/stack/S = W
+
+		if(S.amount >= S.max_amount && !isrobot(user))
```
Comment (JohnGinnane) [inline code/game/objects/items/stacks/stack.dm:242]: Added a nice little message to let the non-robot user know if their stack was already full

**Comment by Carbonhell** [comment]
I don't really like that snowflake check just for cables. There's a variable called parent type which is the path of the parent,for example parent type of /obj/item/stack/cable_coil/random/ would be /obj/item/stack/cable_coil . maybe you could use this ?Or perhaps just scratch the random type of cables and make it be inside the normal cable_coil new proc?

**Comment by JohnGinnane** [comment]
Sounds good, I'll look into using the parent type instead.

As for the random type of cables, how would you go about implementing it into a new procedure?

**Comment by myontasm** [comment]
Fix the snowflake code and I'll give it a +1

**Comment by JohnGinnane** [comment]
Okay I'll take care of that

**Comment by Carbonhell** [comment]
Actually using parent type might fuck up with all other stack types. The best way would be removing the cable coil random types

**Comment by JohnGinnane** [comment]
Can you tell me where the maintenance loot is generated and I can take care of it there?

**Comment by JohnGinnane** [comment]
Okay I'll look at this tonight

File: code/modules/power/cable.dm [hunk 3]
```diff
@@ -484,8 +484,13 @@ var/global/list/datum/stack_recipe/cable_coil_recipes = list ( \
 /obj/item/stack/cable_coil/New(loc, amount = MAXCOIL, var/param_color = null)
 	..()
 	src.amount = amount
+
 	if(param_color)
 		item_color = param_color
+	else
+		item_color = pick("red","yellow","green","blue","pink")
```
Comment (JohnGinnane) [inline code/modules/power/cable.dm:491]: If no colour is specified on cable_coil creation it will pick one at random

**Comment by JohnGinnane** [comment]
Okay dokey, that should be that done. There's no longer  "random" type cable coil. Instead it's random if you create it without specifying the colour.

**Comment by myontasm** [comment]
Good stuff :+1:

**Comment by Carbonhell** [comment]
code's ok,merging

**Pull request merged**